packages/binding-http/src/http-server.ts

```diff
@@ -565,35 +563,33 @@ export default class HttpServer implements ProtocolServer {
             return;
         }
 
-        if (thing.securityDefinitions) {
```

Comment:
Given the comment, this should probably be: `if(thin.security == null)`. Right?

Reply:
As `thing.security` is a mandatory field, I think this part of the code covers the `thing.security.length === 0` case (i.e., the test for `thing.security.length > 0` above does not return true and we did not return from the method early).

This could definitely be a bit clearer, though – maybe we can perform some additional cleanups after this PR has been merged?

Reply:
It is a mandatory field in the TD but not in the `ThingExposeInit`. In my understanding, there is nothing so far to check if the `security` is defined. Probably, due to #911. Currently, we just ask the bindings to handle the `ExposeThingInit` see https://github.com/eclipse-thingweb/node-wot/blob/master/packages/core/src/servient.ts#L38 .  So `thing.security` might be also `null` at that point in time.   Do the new changes take care of this too? 🤔 .

Reply:
Ohh, I see, then this case might not be covered yet! I will double-check this :)

Reply:
Hmm, the parameter passed into the method is of type `security: string | [string, ...string[]]` – so it is mandatory, but I suppose at least one thing that is missing is the case when security is a single string?

Reply:
I tried to provide a fix now in 7a0bd0240de661a9e2df5981038a48cfceb92ba4 – I think that should work for now, however, it seems to me as if the handling of `security` and `securityDefinitions` needs to be revisited in general.

Reply:
Aparently, the code has been correct before, so it reverted the change. Seems to me as if the typing of the `security` field in the `ExposedThing` class/interface might need to be fixed.

Reply:
I know that it is counter-intuitive and I agree that the bug is somewhere else (e.g. #911). If you go upwards in the call hierarchy you'll notice that after producing method nobody checks if `security` is defined, tomorrow I'll try with a practical example. However for the time being we can merge the PR and track this issue in #911 .